### ISSUE TITLE
ThirdPartyLicense for transitive dependencies in glob patterns produce unexpected warnings

### ISSUE THREAD

**Issue by andra2528** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#455]
```xml
<ItemGroup Label="Licenses">
    <ThirdPartyLicense Include="SomeCompanyName.*" Hash="~~~" />
</ItemGroup>
```

Where some package included in the glob `SomeCompanyName.*` references `MongoDB.Libmongocrypt` transitively and leads to the following warning:

```text
Proj0504: The license for MongoDB.Libmongocrypt has changed as its hash is now H+71D1Qif9a+jKUWZKrMdQ
```

Adding the line:

```xml
<ThirdPartyLicense Include="MongoDB.Libmongocrypt" Hash="H+71D1Qif9a+jKUWZKrMdQ" />
```
does not fix the issue.
